### PR TITLE
Add a parameter to `docker` prospector to filter on stream

### PR DESCRIPTION
**Pull request by exekias**
Sometimes you are only interested on stdout or stderr, this parameters
allows to filter the input and only read one of them:

```
prospectors:
  - type: docker
    containers.stream: stderr
    containers.ids:
      - '*'
```

### CODE REVIEW

File: filebeat/prospector/log/config.go [hunk 1]
```diff
@@ -85,7 +85,7 @@ type config struct {
 	JSON         *reader.JSONConfig      `config:"json"`
 
 	// Hidden on purpose, used by the docker prospector:
-	DockerJSON bool `config:"docker-json"`
+	DockerJSON string `config:"docker-json"`
```
Comment (ruflin) [inline filebeat/prospector/log/config.go]: What's the reason for this change?

Comment (exekias) [reply filebeat/prospector/log/config.go]: I'm using the internal parameter to enable docker parsing **and** pass the desired stream through it, possible values:

- `""` - default, docker parsing is disabled
- `all` - default for `docker` prospector, read all streams
- `stdout`
- `stderr`

**Comment by exekias** [comment]
Thank you for the review @ruflin, this should be ready for a second look :smile:

File: filebeat/prospector/log/config.go (hunk 1 above)
Comment (ruflin) [reply filebeat/prospector/log/config.go]: Got it, didn't realise you passed through the stream param.
